libqtile/widget/bluetooth.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2021 Graeme Holliday
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from dbus_next.aio import MessageBus
+from dbus_next.constants import MessageType, BusType
+
+from libqtile.log_utils import logger
+from libqtile.utils import add_signal_receiver
+from libqtile.widget import base
+
+BLUEZ = 'org.bluez'
+BLUEZ_PATH = '/org/bluez/hci0'
+BLUEZ_ADAPTER = 'org.bluez.Adapter1'
+BLUEZ_DEVICE = 'org.bluez.Device1'
+BLUEZ_PROPERTIES = 'org.freedesktop.DBus.Properties'
+
+class Bluetooth(base._TextBox):
+    """
+    Displays bluetooth status or connected device.
+
+    Uses dbus to communicate with the system bus.
+    """
+
+    orientations = base.ORIENTATION_HORIZONTAL
+    defaults = [
+        ('update_interval', 1, 'The update interval.'),
```

Comment:
I think this line is redundant now as you're listening for signals rather than polling regularly.